server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -49,54 +53,77 @@ import           Hasura.Prelude
 import           Hasura.RQL.Types
 import           Hasura.SQL.Value
 
-newtype RemoteRelKey =
-  RemoteRelKey Int
-  deriving (Eq, Ord, Show, Hashable)
-
+-- | Like a 'VQ.Field' but with a path indicating where its response should be
+-- spliced back into the hasura result.
 data RemoteRelField =
```

Comment:
I changed it to `RemoteRelBranch` since it's like part of a tree cut off, but with a path indicating where it fits back into the tree... or something like that...